test/shared/src/main/scala/zio/test/MutableRunnableSpec.scala

```diff
@@ -0,0 +1,144 @@
+package zio.test
+
+import zio.clock.Clock
+import zio.duration._
+import zio.test.environment.TestEnvironment
+import zio.{Chunk, Has, URIO, ZIO, ZLayer}
+
+import scala.util.control.NoStackTrace
+
+/**
+ * Syntax for writing test like
+ * {{{
+ * object MySpec extends MutableRunnableSpec(layer) {
+ *   suite("foo") {
+ *     testM("name") {
+ *     } @@ ignore
+ *
+ *     test("name 2")
+ *   }
+ *   suite("another suite") {
+ *     test("name 3")
+ *   }
+ * }
+ * }}}
+ */
+class MutableRunnableSpec[R <: Has[_]](layer: ZLayer[TestEnvironment, Throwable, R])
+    extends RunnableSpec[TestEnvironment, Any] {
+  self =>
+
+  type ZS = ZSpec[R, Any]
```

Comment:
If the user doesn't need to see this type alias, then I would make it private, so it doesn't clutter up the Scaladoc.

Reply:
I used it only on 3 places so I removed it. And I also need different `ZSpec[Environment, Failure]`